packages/eslint-plugin/tests/rules/sort-imports-metadata/cases.ts

```diff
@@ -0,0 +1,183 @@
+import { convertAnnotatedSourceToFailureCase } from '@angular-eslint/utils';
+import type { MessageIds } from '../../../src/rules/sort-imports-metadata';
+
+const messageId: MessageIds = 'sortImportsMetadata';
+
+export const valid = [
+  `
+  @Component({
+    standalone: true,
+    selector: 'test-component',
+    imports: [aModule, bModule, cModule, dModule],
+  })
+  export class TestComponent { }
+  `,
+  `
+  @Directive({
+    standalone: true,
+    selector: 'test-directive',
+    imports: [aModule, bModule, cModule, dModule],
+  })
+  export class TestDirective { }
+  `,
+  `
+  @Pipe({
+    standalone: true,
+    selector: 'test-pipe',
+    imports: [aModule, bModule, cModule, dModule],
+  })
+  export class TestPipe { }
+  `,
+  `
+  @Component({
+    standalone: true,
+    selector: 'test-component',
+    imports: [
+      aModule,
+      bModule,
+      cModule,
+      dModule
+    ],
+  })
+  export class TestComponent { }
+  `,
+];
+
+export const invalid = [
+  // TODO: This test fails for two reason:
```

Comment:
This `TODO` starts creeping into formatting territory, which we should limit/avoid.